crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_string.rs

```diff
@@ -174,6 +174,10 @@ fn next_useless_escape(str: &str, quote: u8) -> Option<usize> {
                         }
                     }
                     _ => {
+                        // In template literals, \${ is a valid escape for producing a literal ${
+                        if quote == b'`' && c == b'$' && (matches!(it.next(), Some((_, b'{')))) {
+                            return None;
```

Comment:
We should continue to analyze the string. If you return now, you ignore useless escape. For example in `\${\a` `\a` is a useless escape.